File: lagom/examples/bc1-to-bc2-java/bc1-to-bc2-impl/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/ProductService.java
package com.apuex.sales.mapping.bc1ToBc2;

import akka.NotUsed;
import akka.stream.javadsl.Source;
import com.lightbend.lagom.javadsl.api.Descriptor;
import com.lightbend.lagom.javadsl.api.Service;
import com.lightbend.lagom.javadsl.api.ServiceCall;

import java.util.Optional;

import static com.lightbend.lagom.javadsl.api.Service.named;
import static com.lightbend.lagom.javadsl.api.Service.pathCall;

public interface ProductService extends Service {

  public ServiceCall<RetrieveProductCmd, ProductVo> retrieve();

  /**
   * Subscribe from event stream with offset.
   *
   * @param offset timed-uuid or sequence specifies start position
   * @return
   */
  ServiceCall<Source<String, NotUsed>, Source<String, NotUsed>> events(Optional<String> offset);

  @Override
  default Descriptor descriptor() {
    return named("product")
        .withCalls(
            pathCall("/api/events?offset", this::events),
            pathCall("/api/retrieve-product", this::retrieve)
        ).withAutoAcl(true);
  }

}

File: lagom/examples/bc1-to-bc2-scala/api/src/main/scala/com/apuex/sales/mapping/bc1ToBc2/ProductService.scala
package com.apuex.sales.mapping.bc1ToBc2

import javax.inject._

import akka.{Done, NotUsed}
import akka.stream.scaladsl.Source
import com.lightbend.lagom.scaladsl.api.{Descriptor, Service, ServiceCall}
       
trait ProductService extends Service {
  /**
    * Subscribe from event stream with offset.
    *
    * @param offset timed-uuid specifies start position
    * @return
    */
  def events(offset: Option[String]): ServiceCall[Source[String, NotUsed], Source[String, NotUsed]]
         
  def retrieve(): ServiceCall[RetrieveProductCmd, ProductVo]

  
  override final def descriptor: Descriptor = {
    import Service._
  
    named("product")
      .withCalls(
        pathCall("/api/events?offset", events _)
      ).withAutoAcl(true)
  }
       
}
